enterprise/decorators.py

```diff
@@ -134,11 +134,15 @@ def wrapper(request, *args, **kwargs):
         # Enterprise-linked IdP and the pipeline will get them back here.
         if not request.user.is_authenticated():
             parsed_current_url = urlparse(request.get_full_path())
-            parsed_query_string = parse_qs(parsed_current_url.query)
-            parsed_query_string.update({
+            parsed_query_string = parse_qsl(parsed_current_url.query)
+
+            additional_query_parameters = {
                 'tpa_hint': enterprise_customer.identity_provider,
                 FRESH_LOGIN_PARAMETER: 'yes'
-            })
+            }
+            for query_parameter_key, query_parameter_value in additional_query_parameters.iteritems():
+                parsed_query_string.append((query_parameter_key, query_parameter_value))
```

Comment:
use 
```
from six import iteritems
```
and use that to get items iterator for dict.

Reply:
Done